### ISSUE TITLE
Allow additional configuration of vue.config.js

### ISSUE THREAD

**Issue by BickelLukas** [ZachJW34/nx-plus#160]
## Description
When installing vue cli plugins, they sometimes need to be configured via vue.config.js (ex [@vue/cli-plugin-pwa](https://cli.vuejs.org/core-plugins/pwa.html#configuration))

This configuration is not possible right now.

## Motivation
The configuration of these plugins should be possible.

## Suggested Implementation
add a vue.config.js to the application and allow configuration there. This configuration should get merged in with the standard config.

**Comment by BuckyMaler** [comment]
Thanks for creating this issue. I explain my current position on adding `vue.config.js` support [here](https://github.com/ZachJW34/nx-plus/pull/164#pullrequestreview-644089635).

At this time, we want to wait for more user feedback about this before deciding on a direction.

For the example of `@vue/cli-plugin-pwa` that you mentioned, support for configuring that in `workspace.json` can be added. If there's another use case where you've been blocked because `workspace.json` isn't as configurable as `vue.config.js` please explain that. Also, we'd appreciate your contribution if you want to add the `pwa` option. I can provide direction on how to do that if you'd like to work on it.

**Comment by BartInTheField** [comment]
Is vue.config.js support still on the roadmap?

**Comment by westhechiang** [comment]
> Also, we'd appreciate your contribution if you want to add the `pwa` option. I can provide direction on how to do that if you'd like to work on it.

Nice work on the migration to NX 13, I'd like to take a crack at adding the `pwa` option.

**Comment by sang-tesouro** [comment]
I'd like to +1 the `vue.config.js` functionality. I'm currently trying to run a custom loader that runs prior to the `vue-loader`. I cannot seem to find an easy way to achieve this via Webpack configuration, whereas it's a few simple lines in `vue.config.js`.

`// ./vue.config.js
module.exports = {
  chainWebpack: config => {
    config.module
      .rules
      .get('vue')
      .use('components')
      .loader(require.resolve('./custom-loader'))
      .before('vue-loader')
      .end()
  }
}`

If there's an alternative way to achieve this in the current functionality, I'd love to know of it. Otherwise, while I understand that having both `vue.config.js` and `configure-webpack.js` together isn't ideal, it seems worse to throw away a significant portion of Vue's core configuration just because it's a slightly weird process. It's built into Vue and it only makes sense to have it included if we want to achieve Nx Vue parity with other frameworks.

**Comment by westhechiang** [comment]
Yeah, I've spent a ton of hours trying to get `vuetify3` working with nx-plus/vue (https://github.com/ZachJW34/nx-plus/issues/233), but have not been successful, _yet_ 😄.

**Comment by gregveres** [comment]
Where should plugin specific code go? 
For instance, my vue.config.js has the following items in it that I don't know where to put:
```
transpileDependencies: ['vuetify'],
  pwa: {
    name: 'SkyCourt',
    themeColor: '#4c97d2',
    appleMobileWebAppCapable: 'yes'
  },
  productionSourceMap: false,
  pluginOptions: {
    i18n: {
      locale: 'en',
      fallbackLocale: 'en',
      localeDir: 'locales',
      enableInSFC: false
    }
  },
```
It also had devServer which I have successfully moved to project.json.
It also has a publicPath defined like this. I have moved this to project.json too, but  I have yet to tell if this worked properly:

```
publicPath: process.env.NODE_ENV === 'production' ? '/sc/' : '/',
```

I also had two sections for webpack, which I assume I have to figure out how to migrate these into configure-webpack.js:
```
configureWebpack: {
    devtool: 'source-map',
    resolve: {
      alias: {
        '@components': resolve('src/components'),
        '@common': resolve('src/common'),
        '@helpers': resolve('src/helpers'),
        '@pages': resolve('src/pages'),
        '@plugins': resolve('src/plugins'),
        '@store': resolve('src/store'),
        '@locales': resolve('src/locales'),
        '@enums': resolve('src/api/Enums'),
        '@api': resolve('src/api/Interfaces'),
        '@services': resolve('src/api/Services'),
        '@use': resolve('src/use'),
        '@assets': resolve('src/assets'),
        '@vue/composition-api$': '@vue/composition-api/dist/vue-composition-api.mjs'
      }
    },
    plugins: [
      new WebpackAssetsManifest({
        output: 'asset-manifest.json'
      })
      // new webpack.IgnorePlugin({
      //   resourceRegExp: /^\.unit\.ts$/,
      //   contextRegExp: /.*/
      // })
      //, new BundleAnalyzerPlugin()
    ]
  },
  chainWebpack: (config) => {
    // This is here because by default webpack doesn't properly handle const enums
    // in .d.ts or .ts namespaces. This modification of config.module.rule('ts')
    // is needed to get ts-loader to properly handle them.
    // https://github.com/vuejs/vue-cli/issues/2132#issuecomment-431355597
    const rule = config.module.rule('ts');

    rule.uses.delete('thread-loader');
    rule
      // The .test and .exclude are supposed to set webpack to only
      // compile .ts and NOT .test.ts. This config does what I wanted
      // it to output, but that output isn't doing what I expected.
      // I don't understand how files are being picked for ts compilation
      .test(/\.ts$/)
      .exclude.add(/\.test\.ts$/)
      .end()
      .use('ts-loader')
      .loader('ts-loader')
      .tap((options) => {
        options.transpileOnly = false;
        options.happyPackMode = false;

        return options;
      });
  },
```

Any suggestions would be much apprecited. 

This is all Vue 2 as I wait for Vuetify to become fully available for Vue 3.

**Comment by gregveres** [comment]
Oops, I just noticed that there is explicit support for the transpileDependencies already.

**Comment by gregveres** [comment]
I think I figured out how to get the rest of the vue.config.js configuration to work.
You can add a "vue" section to your package.json and that seems to be equivalent to putting stuff in the vue.config.js. 
For instance, I was not able to build once I installed vue-cli-plugin-i18n becuase I kept getting 
```Cannot read properties of undefined (reading 'i18n')```
when trying to build. 
Once I moved this config that used to be in vue.config.js:
```
  pluginOptions: {
    i18n: {
      locale: 'en',
      fallbackLocale: 'en',
      localeDir: 'locales',
      enableInSFC: false
    }
  },
```
into package.json as:
```
  "vue": {
    "pluginOptions": {
      "i18n": {
        "locale": "en",
        "fallbackLocale": "en",
        "localeDir": "locales",
        "enableInSFC": false
      }
    }
  },
```
my error for i18n went away. 

Now I still have other issues that I am working through, but I think all of my config from vue.config.js has been migrated into appropriate locations in the nx config files.

It would be very helpful to have a small doc that explains how to migrate for an existing, working vue 2 app into nx using nx-plus/vue. If I get a chance, after I get my project migrated, I might take a shot at creating one.
